### PR TITLE
Feature/tao 4276 abstract lti service

### PR DESCRIPTION
**Pull request by antoinerobin**

### CODE REVIEW

File: models/classes/class.LtiService.php [hunk 1]
```diff
@@ -19,20 +19,26 @@
...
 
 /**
  * Basic service to handle everything LTI
  * 
  * @author Joel Bout, <joel@taotesting.com>
+ * @deprecated use AbstractLtiService
  */
 class taoLti_models_classes_LtiService extends tao_models_classes_Service 
 {
 	const LIS_CONTEXT_ROLE_NAMESPACE = 'urn:lti:role:ims/lis/';
```
Comment (hutnikau) [inline models/classes/class.LtiService.php]: I guess having the same constants in both services is quite confusing.

Comment (jbout) [inline models/classes/class.LtiService.php]: Is this class referenced in other extensions?

File: models/classes/AbstractLtiService.php [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+ * Copyright (c) 2017 (original work) Open Assessment Technologies SA (under the project TAO-PRODUCT);
+ *
+ */
+
+namespace oat\taoLti\models\classes;
+
+use oat\oatbox\service\ConfigurableService;
+use oat\taoLti\models\classes\LtiMessages\LtiErrorMessage;
+
+abstract class AbstractLtiService extends ConfigurableService
```
Comment (jbout) [inline models/classes/AbstractLtiService.php]: The name is not very specific and no phpdoc block to describe its responsabilities

File: models/classes/class.LtiUser.php [hunk 3]
```diff
@@ -51,7 +53,7 @@ class taoLti_models_classes_LtiUser
 	
 	public function __construct(taoLti_models_classes_LtiLaunchData $ltiLaunchData) {
 	    $this->ltiLaunchData = $ltiLaunchData;
-	    $this->userUri = taoLti_models_classes_LtiService::singleton()->findOrSpawnUser($ltiLaunchData)->getUri();
+	    $this->userUri = $this->getServiceManager()->get(AbstractLtiService::SERVICE_ID)->findOrSpawnUser($ltiLaunchData)->getUri();
```
Comment (jbout) [inline models/classes/class.LtiUser.php]: no calls to getServiceManager in __construct() as there was no opportunity for injection yet

File: models/classes/class.LtiService.php [hunk 4]
```diff
@@ -19,20 +19,26 @@
...
 
 	const LTICONTEXT_SESSION_KEY	= 'LTICONTEXT';
-	
+
+    /** @var AbstractLtiService $ltiService */
+	private $ltiService = null;
+
 	protected function __construct() {
+        $this->ltiService = $this->getServiceLocator()->get(AbstractLtiService::SERVICE_ID);
```
Comment (jbout) [inline models/classes/class.LtiService.php]: no calls to getServiceManager in __construct() as there was no opportunity for injection yet

File: models/classes/class.LtiService.php (hunk 1 above)
Comment (antoinerobin) [inline models/classes/class.LtiService.php]: Yes, ltiDeliveryProvider, ltiBasicOutcome ...

**Pull request closed without merging**